lib/faraday/connection.rb

```diff
@@ -233,20 +233,17 @@ def authorization(type, token)
       set_authorization_header(:authorization, type, token)
     end
 
-    # Internal: Traverse the middleware stack in search of a
-    # parallel-capable adapter.
+    # Internal: Check if the adapter is parallel-capable.
     #
-    # Yields in case of not found.
+    # Yields otherwise or if no adapter is set yet.
     #
-    # Returns a parallel manager or nil if not found.
+    # Returns a parallel manager or nil if yielded.
     def default_parallel_manager
       @default_parallel_manager ||= begin
-        handler = @builder.handlers.detect do |h|
-          h.klass.respond_to?(:supports_parallel?) and h.klass.supports_parallel?
-        end
+        adapter = @builder.adapter.klass if @builder.adapter
 
-        if handler
-          handler.klass.setup_parallel_manager
+        if adapter and adapter.respond_to?(:supports_parallel?) and adapter.supports_parallel?
```

Comment:
You may wish to use `&&` over `and`?

Reply:
Yeah that's the Rubocop convention and I've already planned to scan the entire gem through it before releasing 1.0. However in that file `and` is preferred to `&&` so I went for that.
It's just temporary, we'll eventually converge into the standard and fix all these occurrences later on